nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/LocalServerCommand.java

```diff
@@ -659,4 +656,37 @@ protected String getInstallRootPath() throws CommandException {
         }
         return installRootPath;
     }
-}
+
+    protected HashMap<String, String> getAdditionalTrustandKeyStores() throws IOException, XMLStreamException {
+        HashMap<String, String> additionalTrustandKeyStores = new HashMap<>();
+        try {
+            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+            Document document = documentBuilder.parse(getDomainXml());
+            NodeList jvmOptionsNodes = document.getElementsByTagName("jvm-options");
+
+            for (int i = 0; i < jvmOptionsNodes.getLength(); i++) {
+                if (additionalTrustandKeyStores.containsKey("additionalKeyStores")
+                        && additionalTrustandKeyStores.containsKey("additionalTrustStores")) {
+                    break;
+                }
```

Comment:
I don't think you should exit the loop here.

Additional key & trust stores are defined at the config level, which typically goes like this:
1. server-config
2. default-config
3. extra-config1
4. extra-config2
5. ...

If you defined extra stores in both the server-config (for the DAS) and in extra-config1 (for an instance), you would only be warned about needing to change the stores for those defined in server-config.

Also this has a small flaw in that if you only defined additional key stores, your results would get overwritten. For example, if you defined additional key stores in server-config and default-config, only the additional keystores in default-config would be listed.

So I think you should drop this exit clause and iterate over the entire list, and make sure you append to the value if there's one present (see [Map.computeIfPresent](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfPresent-K-java.util.function.BiFunction-))